### PR TITLE
Modify test case

### PR DESCRIPTION
**Pull request by misgeatgit**
Modified test case in accordance with changes made to STV merge implementation.

### CODE REVIEW

**Comment by inflector** [comment]
Was the merge(stv_1,stv_2) and then merge(stv_2,stv_1) testing to see if the results were the same? Or is merge order dependent?

**Comment by bgoertzel** [comment]
The current merge formula is order-independent.   But order-dependent merge
functions are thinkable in future...

On Tue, Mar 3, 2015 at 10:09 PM, Curtis Faith <notifications@github.com>
wrote:

> Was the merge(stv_1,stv_2) and then merge(stv_2,stv_1) testing to see if
> the results were the same? Or is merge order dependent?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1346#issuecomment-76951276>.
>



-- 
Ben Goertzel, PhD
http://goertzel.org

"The reasonable man adapts himself to the world: the unreasonable one
persists in trying to adapt the world to himself. Therefore all progress
depends on the unreasonable man." -- George Bernard Shaw

File: tests/atomspace/TVMergeUTest.cxxtest [hunk 1]
```diff
@@ -102,31 +102,22 @@ public:
...
-        }
+        //mean=(0.25*0.0 +0.5*1.0)/(0.0+1.0)=0.5 count=0.0+1.0-0.0*0.2=1.0
+        mergedTv = simpleTvs[0]->merge(simpleTvs[1]);
+        TS_ASSERT(mergedTv != simpleTvs[0]);
+        TS_ASSERT(mergedTv->getMean()==0.5f);
+        TS_ASSERT(mergedTv->getCount()==1.0f);
+        mergedTv = simpleTvs[1]->merge(simpleTvs[2]);
+        TS_ASSERT(mergedTv != simpleTvs[1]);
+        //mean=(0.5*1.0 +0.75*2.0)/(1.0+2.0)=0.6667 count=2.0+1.0-1.0*0.2=2.8
+        TS_ASSERT_DELTA(mergedTv->getMean(), 0.6667f,0.0001f);
```
Comment (linas) [inline tests/atomspace/TVMergeUTest.cxxtest]: Please use FLOAT_ACCEPTABLE_ERROR here.  Or maybe call it MEAN_ACCEPTABLE_ERROR

File: tests/atomspace/TVMergeUTest.cxxtest [hunk 2]
```diff
@@ -102,31 +102,22 @@ public:
...
+        TS_ASSERT_DELTA(mergedTv->getCount(),2.8f,0.01f);
```
Comment (linas) [inline tests/atomspace/TVMergeUTest.cxxtest]: Perhaps COUNT_ACCEPTABLE_ERROR here, instead of 0.1

File: tests/atomspace/TVMergeUTest.cxxtest [hunk 3]
```diff
@@ -102,31 +102,22 @@ public:
...
-                    TS_ASSERT(fabs(mergedTv->getCount()  - simpleCounts[i]) <= FLOAT_ACCEPTABLE_ERROR);
-                    TS_ASSERT(fabs(mergedTv->getConfidence() - simpleConfidences[i] ) <= FLOAT_ACCEPTABLE_ERROR);
-                }
-            }
-        }
+        //mean=(0.25*0.0 +0.5*1.0)/(0.0+1.0)=0.5 count=0.0+1.0-0.0*0.2=1.0
+        mergedTv = simpleTvs[0]->merge(simpleTvs[1]);
+        TS_ASSERT(mergedTv != simpleTvs[0]);
+        TS_ASSERT(mergedTv->getMean()==0.5f);
+        TS_ASSERT(mergedTv->getCount()==1.0f);
```
Comment (linas) [inline tests/atomspace/TVMergeUTest.cxxtest]: Based on previous  experience (the MathUTest which passed with gcc 4.6 and failed with gcc 4.7) this result is compiler-dependent.  You may get 0.499999999 someday, and fail the test.  Use MEAN_ACCEPTABLE_ERROR

**Pull request merged**